python/cudf/cudf/pandas/fast_slow_proxy.py

```diff
@@ -204,6 +204,12 @@ def _fsproxy_fast_to_slow(self):
             return fast_to_slow(self._fsproxy_wrapped)
         return self._fsproxy_wrapped
 
+    def get_cudf_pandas_fast_object(self):
```

Comment:
I would recommend that we avoid fast/slow names in the public API. I think `get_cudf_object()` is sufficient. Or possibly `as_cudf()`?

Reply:
Or should we use reserved names like `__as_cudf__()` to make it more obvious that this is a protocol for library developers and not intended for users?

Reply:
We proxy numpy objects too, we will also have to keep that in mind to name these API. How about these names:
1. `__as_fast_object__()`
2. `__as_gpu_object__()`

Reply:
I like (2). Are there any other public APIs involving words like fast/slow or GPU/CPU?

Reply:
I don't think a dunder name is appropriate. Those are typically for protocols that may be implemented across libraries as a standard, not something that single library decides on for itself. I do agree with avoiding fast/slow names, although we may eventually have to rework this if we rip the proxy out of cudf to make it easier for libraries like cuml to reuse. For now `as_gpu_object` seems good to me.

Reply:
The dunder idea was that maybe this should be reserved for other libraries as a documented “protocol” and not presented as a user API. Closer to what IPython does with fancy reprs than the Array API.

Reply:
If you're thinking of IPython's `_repr_html_`, note that that is not a dunder but rather uses single underscores on either side (e.g. [`HTML._repr_html_`](https://github.com/ipython/ipython/blob/a265ad1286acbb8d8c03146eb065defe5a42d312/IPython/core/display.py#L478)). Conversely, the [`__html__` protocol](https://github.com/ipython/ipython/blob/a265ad1286acbb8d8c03146eb065defe5a42d312/IPython/core/display.py#L481) _is_ cross-library and implemented by other libraries (although I don't know exactly how wide the adoption is, I have seen it in a few other places)

Reply:
Good catch! Thanks for the correction.